x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/CompositeAggCursor.java

```diff
@@ -219,7 +219,8 @@ static CompositeAggregationBuilder getCompositeBuilder(SearchSourceBuilder sourc
 
     static boolean couldProducePartialPages(CompositeAggregationBuilder aggregation) {
         for (var agg : aggregation.getPipelineAggregations()) {
-            if (agg instanceof BucketSelectorPipelineAggregationBuilder) {
+            // Use type.equals because there are two copies of BucketSelectorPipelineAggregationBuilder on the classpath
+            if (agg.getType().equals(BucketSelectorPipelineAggregationBuilder.NAME)) {
```

Comment:
> // Use type.equals because there are two copies of BucketSelectorPipelineAggregationBuilder on the classpath

With java modularisation, it shouldn't be possible to have two classes with the same package/name to be exported from different modules?

Reply:
Plugins load them on different classloaders. I'm certainly doing *something* wrong here though.

Reply:
I'm ok with this approach here, since for now there isn't an alternative.